src/comestible_inv_listitem.cpp

```diff
@@ -0,0 +1,771 @@
+        string_format = "";
+    }
+
+    set_print_color( print_color, print_color );
+    return string_format;
+}
+
+void comestible_inv_listitem::set_print_color( nc_color &retval, nc_color default_col )
+{
+    switch( is_selected ) {
+        case SELECTSTATE_NONE:
+            retval = default_col;
+            break;
+        case SELECTSTATE_SELECTED:
+            retval = hilite( c_white );
+            break;
+        case SELECTSTATE_CATEGORY:
+            retval = c_white_red;
+            break;
+        default:
+            break;
+    }
+}
+
+const islot_comestible &comestible_inv_listitem::get_edible_comestible( player &p,
+        const item &it ) const
+{
+    if( it.is_comestible() && p.can_eat( it ).success() ) {
+        // Ok since can_eat() returns false if is_craft() is true
+        return *it.type->comestible;
+    }
+    static const islot_comestible dummy{};
+    return dummy;
+}
+
+
+std::string comestible_inv_listitem::get_time_left_rounded( player &p, const item &it ) const
+{
+    if( it.is_going_bad() ) {
+        return _( "soon!" );
+    }
+
+    time_duration time_left = 0_turns;
+    const time_duration shelf_life = get_edible_comestible( p, it ).spoils;
+    if( shelf_life > 0_turns ) {
+        const double relative_rot = it.get_relative_rot();
+        time_left = shelf_life - shelf_life * relative_rot;
+
+        // Correct for an estimate that exceeds shelf life -- this happens especially with
+        // fresh items.
+        if( time_left > shelf_life ) {
+            time_left = shelf_life;
+        }
+    }
+
+    //commneted code will give approximate times instead of exact ones, in case we want that
+    //const auto make_result = [this](const time_duration& d, const char* verbose_str,
+    //  const char* short_str) {
+    //      return string_format(false ? verbose_str : short_str, time_to_comestible_str(d));
+    //};
+
+    time_duration divider = 0_turns;
+    time_duration vicinity = 0_turns;
+
+    if( time_left > 1_days ) {
+        divider = 1_days;
+        vicinity = 2_hours;
+    } else if( time_left > 1_hours ) {
+        divider = 1_hours;
+        vicinity = 5_minutes;
+    } // Minutes and seconds can be estimated precisely.
+
+    if( divider != 0_turns ) {
+        const time_duration remainder = time_left % divider;
+
+        if( remainder >= divider - vicinity ) {
+            time_left += divider;
+        } else if( remainder > vicinity ) {
+            if( remainder < divider / 2 ) {
+                //~ %s - time (e.g. 2 hours).
+                return string_format( "%s", time_to_comestible_str( time_left ) );
+                //return string_format("> %s", time_to_comestible_str(time_left));
+            } else {
+                //~ %s - time (e.g. 2 hours).
+                return string_format( "%s", time_to_comestible_str( time_left ) );
+                //return string_format("< %s", time_to_comestible_str(time_left));
```

Comment:
Please don't commit commented out code. Either it's useful code and should be run, so it should not be a comment at all. Or it is useless code that should not be there in the first place.

Same below.